### ISSUE TITLE
uptide should be timezone aware

### ISSUE THREAD

**Issue by tkarna** [stephankramer/uptide#1]
As far as I understand uptide does not currently support time zones.
Generally it should be possible for the user to define the initial time as a timezone aware `datetime` object,
```python
timezone = pytz.timezone('Etc/GMT+8')
initial_time = datetime.datetime(2003, 1, 17, 19, 30, tzinfo=timezone)
tide.set_initial_time(init_time)
```
and uptide should do the right thing.

**Comment by stephankramer** [comment]
Yes, I have just looked into this; It's a bit of a rabbit-hole but in the end actually quite easy to implement.

So the summary is: the phases of the harmonically decomposed global model are with respect to the phases of the main astronomical forcing constituents at Greenwich. These we can compute at any point in time using Schwiderski - which gives a formula for these as a function of time with t=0 corresponding to 1975-1-1 0:0:0 UTC(!). So the only place the input datetime is actually used is to compute a difference with this t=0 time - and this is currently using a "naive" datetime without tzinfo. Thus all datetimes in uptide should be though of as being in UTC.

This is easily fixed by providing the t=0 datetime for Schwiderski with a tzinfo=pytz.utc argument. Note however that for most timezones (any one with daylight saving, or any one that has historically changed) the above does not produce a datetime that can be safely used for time differences with other timezones. The above should be:

     timezone = pytz.timezone('US/Pacific')
     initial_time = timezone.localize(datetime.datetime(2003, 1, 17, 19, 30))
     tide.set_initial_time(initial_time)

As far as I understand it, if you use the tzinfo= argument it doesn't correctly take into account the fact that timezone differences are changing over time - and ignores at which time you actually requested the timezone. So for instance if you do tzinfo=tzinfo('Europe/Amsterdam') it will always interpret it as basically UTC+20 min (which was correct for the beginning of the 20th century) regardless of the actual datetime you provided.

**Comment by tkarna** [comment]
OK, sounds good. If we make the t=0 `datetime` object timezone aware, that implies that all datetimes passed to `utide` must be timezone aware (otherwise an error is raised in the subtraction). I think that's a safe thing to do, but it might break some applications. Alternatively, we could detect if the given initial time has a timezone, and if not we assume it to be UTC (and raise a warning perhaps).

Yes, the daylight saving time is a pain. I guess we'd just have to assume that the user can define the timezone correctly. For example, all our Columbia simulations are in PST (Pacific Standard Time, i.e. without daylight saving offset).

**Comment by tkarna** [comment]
FYI according to this,
http://stackoverflow.com/questions/7065164/how-to-make-an-unaware-datetime-timezone-aware-in-python
`pytz.utc.localize(unaware)` applies the correct local time zone taking into account the historical changes or daylight savings.

**Comment by tkarna** [comment]
Thanks, implemented in f855b5b, closing.
